src/topology/mod.rs

```diff
@@ -52,6 +52,8 @@ pub struct RunningTopology {
     detach_triggers: HashMap<String, DisabledTrigger>,
     config: Config,
     abort_tx: mpsc::UnboundedSender<()>,
+    #[cfg(feature = "api")]
+    tap_sinks: weak_table::WeakHashSet<std::sync::Weak<crate::api::tap::TapSink>>,
```

Comment:
I tried to avoid tap sinks bleeding into topology, but hopefully this is slim enough to warrant the approach.

A `weak_table::WeakHashSet` holds weak refs to tap sinks, which are intended to be ephemeral -- they can go out of scope when the client explicitly terminates the subscription, or the WebSocket connection goes away.

The reason to hold refs at all is to reconnect the sinks (which can still be active) across config reloads. This provides an implicitly drop-safe way of doing that.

For situations where things are changed _outside_ of topology, control messages are used instead.